controllers/activemqartemis_reconciler.go

```diff
@@ -620,7 +671,34 @@ func (reconciler *ActiveMQArtemisReconcilerImpl) sourceEnvVarFromSecret(customRe
 	}
 }
 
-func generateAcceptorsString(customResource *brokerv1beta1.ActiveMQArtemis, namer common.Namers, client rtclient.Client) string {
+func (reconciler *ActiveMQArtemisReconcilerImpl) processSSLSecret(secretName string, customResource *brokerv1beta1.ActiveMQArtemis, client rtclient.Client) (*corev1.Secret, error) {
+
+	var trackedSecret *corev1.Secret = nil
+
+	// validate if secret exists
+	secretKey := types.NamespacedName{Name: secretName, Namespace: customResource.Namespace}
+	stringDataMap := map[string]string{}
+	theSecret := secrets.NewSecret(secretKey, stringDataMap, nil)
+	if err := resources.Retrieve(secretKey, client, theSecret); err != nil {
```

Comment:
This initialization could be avoid because the function `Retrieve` also works with empty resources:
```suggestion
	theSecret := &corev1.Secret{}
	if err := resources.Retrieve(secretKey, client, theSecret); err != nil {
```

Reply:
+1